rpcs3/Emu/NP/rpcn_client.cpp

```diff
@@ -1016,6 +1016,43 @@ bool rpcn_client::set_roomdata_internal(u32 req_id, const SceNpCommunicationId&
 	return true;
 }
 
+bool rpcn_client::set_roommemberdata_internal(u32 req_id, const SceNpCommunicationId& communication_id, const SceNpMatching2SetRoomMemberDataInternalRequest* req)
+{
+	std::vector<u8> data{};
+
+	extra_nps::print_set_roommemberdata_int_req(req);
+
+	flatbuffers::FlatBufferBuilder builder(1024);
+	flatbuffers::Offset<flatbuffers::Vector<flatbuffers::Offset<BinAttr>>> final_binattrinternal_vec;
+	if (req->roomMemberBinAttrInternalNum)
+	{
+		std::vector<flatbuffers::Offset<BinAttr>> davec;
+		for (u32 i = 0; i < req->roomMemberBinAttrInternalNum; i++)
+		{
+			auto bin = CreateBinAttr(builder, req->roomMemberBinAttrInternal[i].id, builder.CreateVector(req->roomMemberBinAttrInternal[i].ptr.get_ptr(), req->roomMemberBinAttrInternal[i].size));
+			davec.push_back(bin);
+		}
+		final_binattrinternal_vec = builder.CreateVector(davec);
+	}
+
+	auto req_finished =
+	    CreateSetRoomMemberDataInternalRequest(builder, req->roomId, req->memberId, req->teamId, req->flagFilter, req->flagAttr, final_binattrinternal_vec);
+
+	builder.Finish(req_finished);
+	u8* buf     = builder.GetBufferPointer();
+	usz bufsize = builder.GetSize();
+	data.resize(COMMUNICATION_ID_SIZE + bufsize + sizeof(u32));
+
+	memcpy(data.data(), communication_id.data, COMMUNICATION_ID_SIZE);
+	reinterpret_cast<le_t<u32>&>(data[COMMUNICATION_ID_SIZE]) = static_cast<u32>(bufsize);
+	memcpy(data.data() + COMMUNICATION_ID_SIZE + sizeof(u32), buf, bufsize);
+
+	if (!forge_send(CommandType::SetRoomMemberDataInternal, req_id, data))
```

Comment:
for debugging purposes, it would probably make more sense to assign it to a variable and return that variable

Reply:
wym, callers know what it returned, can be checked there.
also rest of the file already has code like this in 10 places